### PR TITLE
Update examples

### PR DESCRIPTION
**Pull request by wang-boyu**
- update examples folder structure to be consistent with Mesa
- add test case for the examples in the same way as Mesa
- add a geo_schelling_points example

### CODE REVIEW

**Comment by rht** [comment]
I am actually in favor of a flatter file structure hierarchy, instead of example_model/example_model/model.py. There are not that many files anyway. See also: https://github.com/projectmesa/mesa/issues/1269. We can bring this up in the meeting this month.

**Comment by wang-boyu** [comment]
Yup it's probably a matter of personal preferences. No matter what it is, it would be better to be consistent, and updated in the document [here](https://mesa.readthedocs.io/en/latest/best-practices.html#model-layout). We can discuss this later.

**Comment by wang-boyu** [comment]
I've updated the `geo_sir` and `geo_schelling` examples to the flat folder structure, while keeping the `geo_schelling_points` folder structure.

File: examples/geo_schelling_points/geo_schelling_points/agent.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import random
```
Comment (rht) [inline examples/geo_schelling_points/geo_schelling_points/agent.py:1]: This filename needs to be agents.py

Comment (wang-boyu) [reply examples/geo_schelling_points/geo_schelling_points/agent.py]: Updated filename. Changed `geo_sir/agent.py` to `geo_sir/agents.py` too.

**Comment by rht** [comment]
LGTM

**Pull request merged**